src/transformers/modeling_utils.py

```diff
@@ -818,7 +818,6 @@ def _move_model_to_meta(model, loaded_state_dict_keys, start_prefix):
 def _load_state_dict_into_meta_model(
```

Comment:
As explained [here](https://github.com/huggingface/transformers/issues/31970#issuecomment-2325034687), the issue doesn't appear when doing regular loading of the state dict, but only when doing metaloading!